pex/resolve/lockfile/json_codec.py

```diff
@@ -129,6 +131,48 @@ def parse_version_specifier(
                 "The version specifier at '{path}' is invalid: {err}".format(path=path, err=e)
             )
 
+    required_path_mappings = get("path_mappings", dict, optional=True) or {}
+    given_mappings = set(mapping.name for mapping in path_mappings.mappings)
+    unspecified_paths = set(required_path_mappings) - given_mappings
+    if unspecified_paths:
+        raise ParseError(
+            "The lockfile given in {source} requires definition of {required_count} "
```

Comment:
It may be helpful to mention how you set this value. My initial read of the error message made me think that you must set an environment variable. Whereas it sounds like you instead set the command line flag

Reply:
Urg. yeah - I hate non-local coupling like this. the JSON codec should have 0 knowledge of flags. Let me ruminate on being dirty vs maybe a structured exception type the CLI can trap and translate.

Reply:
OK, went with a structured exception type and a message mentioning the flag only in the file that defines the flag.